proposal/LittleLaGi/README.rst

```diff
@@ -0,0 +1,127 @@
+Multi-objective formulations are realistic models for many complex engineering optimization problems. Objectives under consideration usually conflict with each other. Possible solutions form a Pareto set. A Pareto optimal set is a set of solutions that are non-dominated with respect to each other. We can use multi-objective GA to get a set representing the Pareto optimal set well.
+
+
+GA is inspired by evolutionary theory. In GA terminology, a solution vector is called an individual or a chromosome, which normally corresponds to a unique solution x in the solution space. GA operates with a collection of chromosomes, called a population. The population is normally randomly initialized. As the search evolves, the population includes ﬁtter solutions, and eventually, it converges.
+
+This proposal proposes a project named Oh My Gene(OMG), which provides a high-level Python API to construct GA to solve multi-objective problems. Below is a diagram of GA’s general flow:
+
+.. image:: image/GA_general_flow.png
+   :height: 100
+   :width: 200
+   :scale: 50
+   :alt: GA general flow
+
+Prospective Users
+=================
+OMG targets users who need to solve multi-objective problems in specific fields. Users have to map the solution spaces of the problems into forms of chromosomes supported by OMG. After the encoding of a problem, the general GA  can be adapted to solve the transformed problem by tuning some parameters. E.g., generation number, mutation probability, fitness function choice, parent selection method, cross-over method, mutation method, etc.
+
+After the selection of parameters, OMG automatically creates an object of GA and runs it. The result will be shown on the screen for inspection after GA stops. Users can try out different parameter combinations to get the best version of GA that fits their needs by simply altering a few lines of Python code, OMG!
+
+System Architecture
+===================
+| **Tunable Parameters**
+| 1. generation number (default = 100)
+| 2. mating parent ratio (default = 0.5)
+| 3. mutation probability (default = 0.01)
+| 4. fitness function choice (default = weighted sum)
+| 5. parent selection method (default = random selection)
+| 6. cross over method (default = single point crossover)
+| 7. mutation method (default = random mutation)
+|
+| **Output**
+| 1. last population: Show top individuals in the last population.
+| 2. fitness diagram
+.. image:: image/fitness_diagram.png
+   :height: 100
+   :width: 200
+   :scale: 50
+   :alt: fitness diagram
+| 3. new solution rate diagram
+.. image:: image/new_solution_rate_diagram.png
+   :height: 100
+   :width: 200
+   :scale: 50
+   :alt: new solution rate diagram
+|
+| **System Components**
+| 1. **Parser**: Python class. Checks the validity of input parameters and performs necessary transformations.
+| 2. **GA Wrapper**: Python class. Wrapper class for underlying C++ GA class.
+| 3. **Result**: Python class. Contain results returned from GA wrapper.
```

Comment:
How large do you expect the solution (i.e., the problem) to be?

Reply:
I didn't make any assumptions about the problem size.
To the best of my knowledge, users have to alter the parameters to best fit their problems.
Perhaps in most cases, they just inevitably need tons of generations and computing resources?
I'll do more research on GA and try to find the preferable problem size for OMG.

Reply:
You are right that users may choose their own problem size.  Size determines the architecture, though.

For example, if you are shooting for very large problems, the architecture will need to take parallelism into account.  Parallelism is outside the scope of this course.  So this is just an impractical example for you to think about it.

Reply:
I have another class, parallel programming, which also needs to propose a final project.
Currently, I'm planning to parallelize GA, but it also depends on my teammates.
Perhaps I'll have a chance to merge the two projects.

Reply:
It will be wise to leverage multiple courses.  It will be good to add this information in your proposal, and, if you successfully combine two, the final presentation.